api/capi/src/nntrainer.cpp

```diff
@@ -959,6 +959,11 @@ int ml_train_dataset_destroy(ml_train_dataset_h dataset) {
 
   check_feature_state();
 
+  if (dataset == nullptr) {
```

Comment:
maybe I am missing something, but isn't this checked with nntrainer_internal.h L127? if not, then this check will be needed in all the destroy methods.

Reply:
You are correct, this is removed :)